### PR TITLE
velocity-position control transition

### PR DESCRIPTION
**Pull request by Stifael**
This pr fixes position setpoint update after transition from velocity to position control. 
i also resets the position setpoint based on current velocity and max acceleration

### CODE REVIEW

**Comment by Stifael** [comment]
In the past, when switching from position to velocity, it used to have the step-like position setpoint update.
image
With this pr, this step should be gone. 
Furthermore, during switching from velocity to position control, the setpoint was reset to the current position, which causes an overshoot. This behavior was fixed with setting the setpoint according to current velocity and acceleration.

**Comment by LorenzMeier** [comment]
@MaEtUgR Can you review?

**Comment by Stifael** [comment]
with fix:
http://logs.px4.io/plot_app?log=bd9d9d84-9f5c-4493-aea5-e6e78542ab0e
vel-pos-transition-fix

without fix:
http://logs.px4.io/plot_app?log=bd9d9d84-9f5c-4493-aea5-e6e78542ab0e
vel-pos-transition

pleas ignore the bad controller performance when looking at the plots. at some point i will tune the position controller for sitl

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 1]
```diff
@@ -1296,30 +1296,21 @@ MulticopterPositionControl::control_offboard(float dt)
 void
 MulticopterPositionControl::vel_sp_slewrate(float dt)
 {
-	/* limit total horizontal acceleration */
-	math::Vector<2> acc_hor;
-	acc_hor(0) = (_vel_sp(0) - _vel_sp_prev(0)) / dt;
-	acc_hor(1) = (_vel_sp(1) - _vel_sp_prev(1)) / dt;
+	math::Vector<3> acc = (_vel_sp - _vel_sp_prev) / dt;
+	float acc_xy_mag = sqrtf(acc(0) * acc(0) + acc(1) * acc(1));
```
Comment (Stifael) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: there was a bug before: it was _vel_sp -_vel_sp instead of _vel_sp-_vel_sp_prev

**Comment by Stifael** [comment]
I tested this pr with a real vehicle. The setpoint is reset now correctly:
before:
setpoint_reset_old
new:
setpoint-new

If the parameters hold_z and hold_xy are set to  small numbers, then the change in position_setpoint when entering position-control from velocity-control is not very significant. However, setting those parameters too small increases the risk that position-control will never be entered. With this pr, the thresholds can be increased without overshooting the setpoint too much.

**Comment by jgoppert** [comment]
Nice!

**Pull request merged**